pygmt/src/triangulate.py

```diff
@@ -0,0 +1,389 @@
+"""
+triangulate - Delaunay triangulation or Voronoi partitioning and gridding of
+Cartesian data.
+"""
+import warnings
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+class triangulate:  # pylint: disable=invalid-name
+    """
+    Delaunay triangulation or Voronoi partitioning and gridding of Cartesian
+    data.
+
+    Triangulate reads in x,y[,z] data and performs Delaunay triangulation,
+    i.e., it finds how the points should be connected to give the most
+    equilateral triangulation possible. If a map projection (give *region* and
+    *projection*) is chosen then it is applied before the triangulation is
+    calculated. By default, the output is triplets of point id numbers that
+    make up each triangle. The id numbers refer to the points position (line
+    number, starting at 0 for the first line) in the input file. If **outgrid**
+    and **spacing** are set a grid will be calculated based on the surface
+    defined by the planar triangles. The actual algorithm used in the
+    triangulations is either that of Watson [1982] [Default] or Shewchuk [1996]
+    (if installed; type **gmt triangulate -** on the command line to see which
```

Comment:
The GMT docs say the default is Watson, but I think the conda-forge package compiles with Shewchuk instead (see the error log in https://github.com/conda-forge/gmt-feedstock/pull/186#issuecomment-986186576)? Also, I tried running `gmt triangulate -` on the command line but didn't see anything about Watson/Shewchuk...

Reply:
Confusing. I think that the proper way to find out is `gmt get GMT_TRIANGULATE`. Feel free to open an upstream issue requesting better documentation.

Reply:
I found a note at https://github.com/GenericMappingTools/gmt/blob/6.3.0/src/TRIANGLE.HOWTO that the Watson [1982] algorithm is the GMT default because it is public domain, whereas the Shewchuk [1996] algorithm is under copyright (see https://github.com/GenericMappingTools/gmt/blob/6.3.0/src/README.TRIANGLE) and requires a license for commercial use.

I'll modify the docs here in PyGMT to say that `gmt get GMT_TRIANGULATE` is the way to check, and have a look at the build code at https://github.com/conda-forge/gmt-feedstock to see if I can find the setting.

Reply:
Upstream issue opened at https://github.com/GenericMappingTools/gmt/issues/6436.

Reply:
Ok, so the default in GMT appears to be Shewchuk [1996] instead of Watson [1982], so I've modified the docstring in 02ac1d8dc0e5328ec784625ea92304976d8739da.